src/main/java/com/github/redouane59/twitter/helpers/RequestHelper.java

```diff
@@ -1,119 +1,56 @@
-      RequestBody requestBody = RequestBody.create(json, MediaType.parse("application/json"));
-      Request request = new Request.Builder()
-          .url(httpBuilder.build())
-          .post(requestBody)
-          .build();
-      Response response = this.httpClient
-          .newCall(request).execute();
-      String stringResponse = response.body().string();
-      if (response.code() < 200 || response.code() > 299) {
-        logApiError("POST", url, stringResponse, response.code());
-      }
-      if (classType.equals(String.class)) { // dirty, to manage token oauth1
-        result = (T) stringResponse;
-      } else {
-        result = TwitterClient.OBJECT_MAPPER.readValue(stringResponse, classType);
-      }
-    } catch (Exception e) {
-      LOGGER.error(e.getMessage(), e);
-    }
-    return Optional.ofNullable(result);
-
+    return makeRequest(Verb.POST, url, parameters, requestBodyJson, true, classType);
   }
 
   public <T> Optional<T> postRequest(String url, Map<String, String> parameters, Class<T> classType) {
     return postRequestWithBodyJson(url, parameters, null, classType);
   }
-
+  
   public <T> Optional<T> uploadMedia(String url, File file, Class<T> classType) {
-    T result = null;
-    try {
-
-      HttpUrl.Builder httpBuilder = HttpUrl.parse(url).newBuilder();
-      RequestBody requestBody = new MultipartBody.Builder().setType(MultipartBody.FORM)
-                                                           .addFormDataPart("media", file.toString(),
-                                                                            RequestBody.create(MediaType.parse("application/octet-stream"),
-                                                                                               file))
-                                                           .build();
-      Request request = new Request.Builder()
-          .url(httpBuilder.build())
-          .post(requestBody)
-          .build();
+	try {
+		return uploadMedia(url, file.getName(), Files.readAllBytes(file.toPath()), classType);
+	} catch (IOException e) {
+		LOGGER.error(e.getMessage(), e);
+		return Optional.empty();
+	}
+  }
 
-      Response response       = this.httpClient.newCall(request).execute();
-      String   stringResponse = response.body().string();
-      if (response.code() < 200 || response.code() > 299) {
-        logApiError("POST", url, stringResponse, response.code());
-      }
-      result = TwitterClient.OBJECT_MAPPER.readValue(stringResponse, classType);
-    } catch (Exception e) {
-      LOGGER.error(e.getMessage(), e);
-    }
-    return Optional.ofNullable(result);
+  public <T> Optional<T> uploadMedia(String url, String fileName, byte[] data, Class<T> classType) {
+    OAuthRequest request = new OAuthRequest(Verb.POST, url);
+    request.addBodyPartPayloadInMultipartPayload(new FileByteArrayBodyPartPayload("application/octet-stream", data, "media", fileName));
```

Comment:
Do you manage to make it work ? 
Calling twitterClient.uploadMedia(new File("C:\\Users\\Perso\\Pictures\\bot.PNG"), MediaCategory.TWEET_IMAGE) I have a null pointer exception :/

Reply:
Moreover, the `testAddAndDeleteFilteredStreamRules` in ITwitterClientV2Test is not passing anymore. It looks like `addFilteredStreamRule` method is broken. Can you check too ?

Reply:
Please check latest commit.

Reply:
Yes with the latest commit both are now working :)